base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_cvd.cc

```diff
@@ -475,6 +491,27 @@ Result<void> SaveConfig(FetcherConfig& config,
   return {};
 }
 
+Result<std::vector<std::string>> ExtractImageContents(
+    const std::string& image_filepath, const std::string& target_dir,
+    const bool keep_archive) {
+  std::vector<std::string> files;
+  if (IsDirectory(image_filepath)) {
+    // The image is already uncompressed. Link or move its contents.
+    if (keep_archive) {
+      // Must use hard linking due to the way fetch_cvd uses the cache.
+      files = CF_EXPECT(
+          HardLinkDirecoryContentsRecursively(image_filepath, target_dir));
+    } else {
+      files = CF_EXPECT(MoveDirectoryContents(image_filepath, target_dir));
+      RecursivelyRemoveDirectory(image_filepath);
```

Comment:
This isn't necessary after a `move`. Only the top level directory should remain, so you could use `rmdir(2)`

Reply:
Done.